tests/Signals_perftest2.m

```diff
@@ -0,0 +1,161 @@
+        @plus, @minus, @times, @rdivide, @onValue, @map, @scan, @subscriptable} % functions to test
+    NumNodes = {30 100 500 1000}; % number of nodes in the network, not counting A & B
+     OnlyBasicOps = {1 0}; % flag for whether or not to only use logical/scalar ops in the network tested in 'test_networkOps'
+    BasicOps = {@gt, @ge, @lt, @le, @eq,...
+        @plus, @minus, @times, @rdivide} % logical/arithmetical subset of functions to test
+  end
+  
+  methods (TestMethodSetup)
+    
+    function createNetwork(testCase)
+      % creates the signals network in which we will run the tests
+      testCase.Net = sig.Net;
+      testCase.A = testCase.Net.origin('A');
+      testCase.B = testCase.Net.origin('B');
+    end
+    
+  end
+    
+  methods (Test)
+    
+    function test_singleOps(testCase, Ops)
+      % benchmarks operations on a single signal without propogations
+      
+      % set-up op to test
+      switch func2str(Ops)
+        % logical and scalar ops
+        case {'gt', 'ge', 'lt', 'le', 'eq', 'plus', 'minus', 'times', 'rdivide'}
+          c = feval(Ops, testCase.A, testCase.B); %#ok<*NASGU>
+          post(testCase.B, 2);
+        % listener ('onValue') & 'map' ops
+        case {'onValue', 'map'}
+          f = @plus;
+          c = feval(Ops, testCase.A, @(x) f(1,x));
+          while (testCase.keepMeasuring)
+            post(testCase.A, 1);
+          end
+        % 'scan'
+        case {'scan'}
+          f = @plus;
+          c = feval(Ops, testCase.A, f, testCase.B);
+          post(testCase.B, 0);
+          while (testCase.keepMeasuring)
+            post(testCase.A, 1);
+          end
+        % 'subscriptable'
+        case {'subscriptable'}
+          s = struct('A', 1);
+          c = feval(Ops, testCase.A);
+          c_A = c.A;
+          while (testCase.keepMeasuring)
+            post(testCase.A, s);
+          end
+      end
+      
+      % test op
+      switch func2str(Ops)
+        case {'nop', 'post'}
+          while (testCase.keepMeasuring)
+            for i = 1:testCase.Reps
+              feval(Ops, testCase.A, 1);
+            end
+          end
+        case {'gt', 'ge', 'lt', 'le', 'eq', 'plus', 'minus', 'times',...
+            'rdivide', 'onValue', 'map', 'scan'}
+          while (testCase.keepMeasuring)
+            for i = 1:testCase.Reps
+              post(testCase.A, 1); % this post will trigger the update of signal 'c', defined above
+            end
+          end
+        case {'subscriptable'}
+          while (testCase.keepMeasuring)
+            for i = 1:testCase.Reps
+              post(testCase.A, s); % this post will trigger the update of signal 'c_A', defined above
+            end
+          end
+      end
+      
+    end
+    
+    function test_networkOps(testCase, OnlyBasicOps, NumNodes)
+      % benchmarks operations on a signals network upon posting to a signal
+      % which updates all other signals in the network
```

Comment:
According to Chris, this setup isn't very informative.  The depth of the network impacts the performance, not the number of nodes per se.  Perhaps I'm understanding this incorrectly, but aren't you just creating lots of nodes that all just depend on the same one Signal?

Reply:
what exactly do you mean by depth? As in, how could we quantify depth?

> aren't you just creating lots of nodes that all just depend on the same one Signal?

Yes, essentially. I wanted to show that there's not a significant drop-off in performance the more "listeners" you have for one "event", which may seem intuitive, but is also nice to confirm.

Reply:
I also realized the "depth" vs "width" testing has some to do with the cpu architecture and it's ability to parallelize operations. E.g. the more logical processors a CPU has the much better it should perform on width testing (I tested this out while accounting for baseline cpu-speed on 2 vs. 4 processor CPUs in lab). 

So I think width can still impact performance, particularly on CPUs which don't optimize parallelization, and it's important to test for both, which I've done in the latest 'perf_test'